### ISSUE TITLE
Parsing a JSON array with small number returns an array with an infinity float value

### ISSUE THREAD

**Issue by plokhotnyuk** [ngs-doo/dsl-json#108]
Code to reproduce with Scala:
```
import com.dslplatform.json.DslJson

val dslJson = new DslJson[Any](new DslJson.Settings[Any]())
val reader =  dslJson.newReader().process("[7.006492321624086e-46]".getBytes, 23)
reader.read()
println(dslJson.tryFindReader(classOf[Array[Float]]).read(reader)(0))
```

**Comment by zapov** [comment]
Tnx for the bug report.
I'm not too happy with current float and double parsing code. It should be rewritten using some known algorithms.

**Comment by zapov** [comment]
Fix will be included in v1.8.5

**Comment by zapov** [comment]
v1.8.5 released
